File: 프로그래머스/lv0/120846. 합성수 찾기/합성수 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n)
{
    int answer = 0;

    for (int i = 4; i <= n; i++)
    {
        if (i % 2 == 0 || i % 3 == 0)
            answer++;
        else if (i >= 25 && i % 5 == 0)
            answer++;
        else if (i >= 49 && i % 7 == 0)
            answer++;
   
    }
    return answer;
}

File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for (int i = 4; i <= n; i++) {
            if (check(i)) answer++;
        }
        
        return answer;
    }
    
    private boolean check(int n) {
        int count = 1;
        for (int i = 2; i <= n; i++) {
            if (n % i == 0) {
                count++;
            }
            
            if (count >= 3) return true;
        }
        return false;
    }
}